Hardcode packages variable in setup.py to prevent installation of files in 'tests' directory

AndrewAmmerlaan opened the pull request:
Hello,

I've ran into a small problem with the setup.py file.  Portage (gentoo's package manager) kept complaining about my ebuild:

```
>>> Failed to emerge dev-python/spotify-music-videos-1.8, Log file:

>>>  '/var/tmp/portage/dev-python/spotify-music-videos-1.8/temp/build.log'

* Messages for package dev-python/spotify-music-videos-1.8:

* ERROR: dev-python/spotify-music-videos-1.8::localrepo failed (install phase):
*   Package installs 'tests' package which is forbidden and likely a bug in the build system.
* 
* Call stack:
*     ebuild.sh, line  125:  Called src_install
*   environment, line 3423:  Called distutils-r1_src_install
*   environment, line 1119:  Called _distutils-r1_run_foreach_impl 'distutils-r1_python_install'
*   environment, line  466:  Called python_foreach_impl 'distutils-r1_run_phase' 'distutils-r1_python_install'
*   environment, line 2970:  Called multibuild_foreach_variant '_python_multibuild_wrapper' 'distutils-r1_run_phase' 'distutils-r1_python_install'
*   environment, line 2284:  Called _multibuild_run '_python_multibuild_wrapper' 'distutils-r1_run_phase' 'distutils-r1_python_install'
*   environment, line 2282:  Called _python_multibuild_wrapper 'distutils-r1_run_phase' 'distutils-r1_python_install'
*   environment, line  803:  Called distutils-r1_run_phase 'distutils-r1_python_install'
*   environment, line 1087:  Called distutils-r1_python_install
*   environment, line 1005:  Called die
* The specific snippet of code:
*               die "Package installs '${p}' package which is forbidden and likely a bug in the build system.";
```

I was a bit confused about this, as I also wrote ebuilds for some dependencies (lyricwikia, python-vlc) which also have a 'tests' directory, yet portage was absolutely fine with these ebuilds.

In my initial ebuild I fixed this by just patching out the whole 'tests' directory prior to installing. However, this solution is imperfect so I kept looking for what was differentiating this repository from the lyricwikia one.

It took some time but I finally found the problem, in setup.py. The line `packages=find_packages(),` causes setuptools to find all directories, including 'tests', (which shouldn't be installed). Hardcoding it to `['spotify-videos']` similarly to the setup.py's I found in other repositories fixes the problem.

[EDIT]: See also: https://github.com/gentoo/gentoo/pull/13870

marioortizmanero commented:
Thanks for your contribution :)) I initially just used a lazy find_packages() but yeah, hardcoding them is a better idea since it's just one.

marioortizmanero commented on setup.py:
```diff
@@ -22,7 +22,7 @@
     name='spotify-videos',
     version=__version__,
-    packages=find_packages(),
+    packages=['spotify-videos']
```
missing a `,` at the end

AndrewAmmerlaan commented:
> it seems you missed a , after the packages parameter. Can you fix that so that I can merge the PR?

Aah I missed that one, that's what I get for being lazy and just copying the change instead of applying the patch I pushed to my branch of the Gentoo repo earlier. Thanks for catching that

marioortizmanero commented:
Sorry, one last thing: the module name is `spotify_videos`, not `spotify-videos`.

AndrewAmmerlaan commented:
> Sorry, one last thing: the module name is spotify_videos, not spotify-videos.

Oops, sorry about that, I had it correct in the patch I uploaded to the Gentoo repo, but then made a mistake copying it (again). 

I think there's a lesson for me here: always apply the patch that has been proven to work instead of being lazy and thinking I can get away with copying the changes ;D

marioortizmanero commented:
Don't worry it works perfectly now :)

marioortizmanero commented:
@AndrewAmmerlaan I've released the 1.8.1 version with this hotfix, in case you wanted to download the `.tar.gz` file for the Gentoo installation: https://github.com/marioortizmanero/spotify-music-videos/releases/tag/1.8.1

AndrewAmmerlaan commented:
> 

@AndrewAmmerlaan I've released the 1.8.1 version with this hotfix, in case you wanted to download the .tar.gz file for the Gentoo installation: https://github.com/marioortizmanero/spotify-music-videos/releases/tag/1.8.1

Awesome, thank you, I've updated the ebuild and removed the patch form my [pull request](https://github.com/gentoo/gentoo/pull/13870) in the gentoo repository
